zerver/lib/markdown/fenced_code.py

```diff
@@ -472,8 +473,19 @@ def format_quote(self, text: str) -> str:
         for paragraph in paragraphs:
             lines = paragraph.split("\n")
             quoted_paragraphs.append("\n".join("> " + line for line in lines))
+            quoted_paragraphs = [self.format_quoted_emoji(quote) for quote in quoted_paragraphs]
         return "\n".join(quoted_paragraphs)
 
+    # Used when quoting a message with an emoticon
+    # to prevent its translation to an emoji
+    # and preserve the original message context
+    def format_quoted_emoji(self, text: str) -> str:
```

Comment:
This function should be changed, probably to something: `escape_emoji_inside_quote`?
Because function with the name: `format_*` implies that they do formatting for some `lang` groups like `quote`, `spoiler`, `code`.